### PR TITLE
improve upgrade docs

### LINKED ISSUE #4330: [Documentation] Split upgrade docs
Split the upgrade doc so it is more clear what should be done.

one doc for Core-2 -> Core-3

a second doc for Core-3.0.x -> Core-3.0.x+1

### PR DESCRIPTION
**Pull request by craigh**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| New feature?      | no
| BC breaks?        | no
| Deprecations?     | no
| Fixed tickets     | fixed #4330
| Refs tickets      | -
| License           | MIT
| Changelog updated | no

## Description

improve upgrade docs

### CODE REVIEW

File: docs/Setup/Upgrade.md [hunk 1]
```diff
@@ -1,67 +1,60 @@
...
-  - Note 2: If you have `mod_suexec` installed for Apache the CLI will run into permission problems. (If you are not sure 
-    check your phpinfo.) `mod_suexec` often is used in shared hosting environments. In this case, the CLI installer is not 
-    recommended, please use the Web Installer.
-- Copy your previous installation's `/app/config/custom_parameters.yml` to `/config/services_custom.yaml` in your new installation.
-  - note the name change and the suffix changed from `.yml` to `.yaml`
+  - If you have `mod_suexec` installed for Apache the CLI will run into permission problems. (If you are not sure 
+    check your phpinfo.) `mod_suexec` is often used in shared hosting environments. In this case, the CLI installer is not 
+    recommended. Instead, please use the Web Installer.
+- Please note, your _web root_ **must** now point to the `public/` directory. Adjust your symlinks or directory structure as needed.
+  (see https://symfony.com/doc/current/setup/web_server_configuration.html for more information).
```
Comment (Guite) [inline docs/Setup/Upgrade.md:40]: :+1:

File: docs/Setup/Upgrade.md [hunk 2]
```diff
@@ -1,67 +1,60 @@
...
-- Return _compatible_ extensions to the `src/extensions` directory.
-  - **DO NOT copy the old Profile and Legal module** as new versions of these are provided, and their location may differ.
-- Copy your backup contents of `/userdata` (1.x) or `/web/uploads` (2.x) into `/public/uploads`
+- Copy old files to _new_ locations:
+    - Copy your previous installation's `/app/config/custom_parameters.yml` to `/config/services_custom.yaml` in your new installation.
+        - note the name change and the suffix changed from `.yml` to `.yaml`
+    - Copy your _compatible_ custom theme to the `src/extensions` directory (There is no `/themes` directory any longer).
+        - After upgrade, the site will default to the ZikulaBootstrapTheme until you complete testing of your custom theme.
+    - Return _compatible_ modules (extensions) to the `src/extensions` directory.
+        - **DO NOT copy the old Profile and Legal module** as new versions of these are provided, and their location may differ.
```
Comment (Guite) [inline docs/Setup/Upgrade.md:50]: Maybe add: `If unsure about whether a module is compatible of whether a new compatible version is available please ask for support in our Slack channels.` ?

File: docs/Setup/UpgradeFrom3to3.x.x.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+## Upgrading
+- Before uploading the new files, delete (or move) **all files** in your web root (typically `public_html` or `httpdocs`).
+- Upload the new package and unpack the archive.
+  - Please read the [INSTALL docs](INSTALL.md#upload) for detailed information on proper uploading.
+  - Ensure `/var/cache` and `/var/log` directories are writable both by the web server and the command line user.
+  - If you have `mod_suexec` installed for Apache the CLI will run into permission problems. (If you are not sure 
+    check your phpinfo.) `mod_suexec` often is used in shared hosting environments. In this case, the CLI upgrader is not 
+    recommended, please use the Web Upgrade.
+- Copy old files to parallel locations:
+    - Copy your previous installation's `/.env.local` to `/.env.local` in your new installation.
```
Comment (Guite) [inline docs/Setup/UpgradeFrom3to3.x.x.md:32]: Maybe add: `Do the same for any `/.env_{env}.local` file which may exist in addition.` ?

Comment (craigh) [reply docs/Setup/UpgradeFrom3to3.x.x.md:32]: like what? the test envs are not needed. there are no others.

**Pull request merged**